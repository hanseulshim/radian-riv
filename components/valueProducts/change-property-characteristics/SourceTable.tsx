import React, { useState, useCallback } from 'react'
import Table from 'components/common/Table'
import Radio from 'components/common/Radio'

interface Props {
  tableData: any[]
  setSelectedSource: (source: string) => void
  selectedSource: string
}

export default function SourceTable({
  tableData,
  setSelectedSource,
  selectedSource
}: Props) {
  const [data, setData] = useState([])
  const [columns, setColumns] = useState([])

  const fetchData = useCallback(async () => {
    const width = 85
    setColumns([
      {
        Header: ' ',
        width: 50,
        align: 'center',
        accessor: 'input'
      },
      {
        Header: 'SELECT SOURCE ',
        width: 250,
        align: 'left',
        accessor: 'source'
      },
      {
        Header: 'Bed',
        width,
        align: 'left',
        accessor: 'bed'
      },
      {
        Header: 'Bath',
        width,
        align: 'left',
        accessor: 'bath'
      },
      {
        Header: 'SqFt',
        width,
        align: 'left',
        accessor: 'sqft'
      },
      {
        Header: 'Units',
        width,
        align: 'left',
        accessor: 'units'
      },
      {
        Header: 'Garage',
        width,
        align: 'left',
        accessor: 'garage'
      },
      {
        Header: 'Lot Size',
        width,
        align: 'left',
        accessor: 'lotSize'
      },
      {
        Header: 'Yr. Built',
        width,
        align: 'left',
        accessor: 'yearBuilt'
      }
    ])
    setData(tableData)
  }, [tableData])

  return (
    <Table
      columns={columns}
      data={data.map(row => {
        return {
          input: (
            <Radio
              onChange={e => setSelectedSource(e.target.value)}
              value={row.source}
              checked={selectedSource === row.source}
              label={''}
            />
          ),
          ...row
        }
      })}
      fetchData={fetchData}
      width={1000}
    />
  )
}
