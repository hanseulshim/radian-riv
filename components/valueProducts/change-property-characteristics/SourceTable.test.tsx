import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SourceTable from './SourceTable'

describe('SourceTable', () => {
  let component
  beforeAll(() => {
    const props = {
      tableData: [],
      setSelectedSource: () => {},
      selectedSource: 'User'
    }
    component = shallow(<SourceTable {...props} />)
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
