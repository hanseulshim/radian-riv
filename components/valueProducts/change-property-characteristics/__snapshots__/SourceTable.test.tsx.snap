// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SourceTable should match snapshot 1`] = `
<Table
  columns={Array []}
  data={Array []}
  fetchData={[Function]}
  width={1100}
/>
`;
